pkg/datapath/loader/compile.go

```diff
@@ -447,3 +456,38 @@ func compileNetwork(ctx context.Context) error {
 	}
 	return nil
 }
+
+// compileOverlay compiles BPF programs in bpf_overlay.c.
+func compileOverlay(ctx context.Context, opts []string) error {
+	dirs := directoryInfo{
+		Library: option.Config.BpfDir,
+		Runtime: option.Config.StateDir,
+		Output:  option.Config.StateDir,
+		State:   option.Config.StateDir,
+	}
+	scopedLog := log.WithField(logfields.Debug, true)
+
+	versionCmd := exec.CommandContext(ctx, compiler, "--version")
+	compilerVersion, err := versionCmd.CombinedOutput(scopedLog, true)
+	if err != nil {
+		return err
+	}
+	versionCmd = exec.CommandContext(ctx, linker, "--version")
+	linkerVersion, err := versionCmd.CombinedOutput(scopedLog, true)
+	if err != nil {
+		return err
+	}
+	scopedLog.WithFields(logrus.Fields{
+		compiler: string(compilerVersion),
+		linker:   string(linkerVersion),
+	}).Debug("Compiling overlay programs")
+
+	overlayTcProg.Options = opts
```

Comment:
This is a write to a package global and will need a copy.

Reply:
Went ahead and removed the package global all together, no need to have it in the first place.